### PR TITLE
Hash tooling changes

### PR DESCRIPTION
**Pull request by tiehuis**
Update the benchmarking scripts for hash/crypto functions. Adds some new hashes to them and small key throughput tests.

This gives moderate speed improvements when hashing small keys.
The crc/adler/fnv inlining did not provide enough speed up to warrant
the change.

Also, I've inlined some hashes since the speed seems better.

```
OLD:

wyhash
  small keys: 2277 MiB/s [c14617a1e3800000]
siphash(1,3)
  small keys:  937 MiB/s [b2919222ed400000]
siphash(2,4)
  small keys:  722 MiB/s [3c3d974cc2800000]
fnv1a
  small keys: 1580 MiB/s [70155e1cb7000000]
adler32
  small keys: 1898 MiB/s [00013883ef800000]
crc32-slicing-by-8
  small keys: 2323 MiB/s [0035bf3dcac00000]
crc32-half-byte-lookup
  small keys:  218 MiB/s [0035bf3dcac00000]

NEW:

wyhash
  small keys: 2775 MiB/s [c14617a1e3800000]
siphash(1,3)
  small keys: 1086 MiB/s [b2919222ed400000]
siphash(2,4)
  small keys:  789 MiB/s [3c3d974cc2800000]
fnv1a
  small keys: 1604 MiB/s [70155e1cb7000000]
adler32
  small keys: 1856 MiB/s [00013883ef800000]
crc32-slicing-by-8
  small keys: 2336 MiB/s [0035bf3dcac00000]
crc32-half-byte-lookup
  small keys:  218 MiB/s [0035bf3dcac00000]
```

### CODE REVIEW

File: std/hash/crc.zig [hunk 1]
```diff
@@ -9,17 +9,17 @@ const std = @import("../std.zig");
...
+    Koopman = 0xeb31d82e,
 };
 
 // IEEE is by far the most common CRC and so is aliased by default.
-pub const Crc32 = Crc32WithPoly(Polynomial.IEEE);
+pub const Crc32 = Crc32WithPoly(.IEEE);
 
 // slicing-by-8 crc32 implementation.
-pub fn Crc32WithPoly(comptime poly: u32) type {
+pub fn Crc32WithPoly(comptime poly: Polynomial) type {
```
Comment (daurnimator) [inline std/hash/crc.zig]: Could you undo this change? It's not uncommon to pass a custom Crc32 polynomial. See https://en.wikipedia.org/wiki/Cyclic_redundancy_check#Polynomial_representations_of_cyclic_redundancy_checks for a sample of the huge range in common use.

Comment (tiehuis) [reply std/hash/crc.zig]: Sure, that is something I overlooked when I made this change. Probably why I originally did it this way originally.

Comment (tiehuis) [reply std/hash/crc.zig]: Made a PR at #3118.
